core/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala

```diff
@@ -216,6 +216,34 @@ class SQLPlanParserSuite extends BaseTestSuite {
     }
   }
 
+  runConditionalTest("test subexecutionId mapping to rootExecutionId",
+    subExecutionSupportedSparkGTE340) {
+    val eventlog = ToolTestUtils.getTestResourcePath("" +
+        "spark-events-qualification/db_subExecution_id.zstd")
+    val app = createAppFromEventlog(eventlog)
+    // Get sum of durations of all the sqlIds. It contains duplicate values
+     val totalSqlDuration = app.sqlIdToInfo.values.map(x=> x.duration.getOrElse(0L)).sum
+
+    // This is to group the sqlIds based on the rootExecutionId. So that we can verify the
+    // subExecutionId to rootExecutionId mapping.
+     val rootIdToSqlId = app.sqlIdToInfo.groupBy { case (_, info) =>
+      info.rootExecutionID
+    }
+    assert(rootIdToSqlId.get(Some(5L)).map(_.keys.toSeq.sorted) == Some(Seq(5,6,7,8,9,10)))
```

Comment:
Assuming the key `Some(5L)` will always be present in `rootIdToSqlId` and expected keys are `5,6,7,8,9,10`, can we simplify this as:

```suggestion
    assert(rootIdToSqlId(Some(5L)).keys.toSet == Set(5,6,7,8,9,10))
```

(as ordering does not matter)